compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -671,6 +671,9 @@ pub enum TerminatorKind<'tcx> {
     /// as parameters, and `None` for the destination. Keep in mind that the `cleanup` path is not
     /// necessarily executed even in the case of a panic, for example in `-C panic=abort`. If the
     /// assertion does not fail, execution continues at the specified basic block.
+    ///
+    /// When overflow checking is disabled and we are generating run-time code, the `Overflow*`
+    /// variants of this terminator are codegened as simple `goto target`.
```

Comment:
This is very suspicious. The spec of a language should never mention "codegen".  Shouldn't this just say "when overflow checking is disabled, the `Overflow*` variants behave as `goto target`"? And then the interpreter should also implement those semantics.

Reply:
Ah, this is partially copied from the existing CheckedBinaryOp docs, but the partial copy made it even more confusing. That existing doc should also be adjusted...
```suggestion
    /// When overflow checking is disabled and this is run-time MIR (as opposed to
    /// compile-time MIR that is used for CTFE), the `Overflow*`
    /// variants of this terminator behave as `goto target`.
```
Having semantic differences between runtime and compiletime MIR is not great but is not new either.